tests/test_lora_conversion.py

```diff
@@ -0,0 +1,513 @@
+        # with a fixed rank, we expect target_modules to be set on the LoRA config but not rank_pattern, alpha_pattern
+        lora_config, _ = convert_to_lora(lokr_model, rank=8)
+        assert isinstance(lora_config, LoraConfig)
+        assert lora_config.r == 8
+        assert lora_config.lora_alpha == 8
+        assert lora_config.target_modules
+        assert not lora_config.rank_pattern
+        assert not lora_config.alpha_pattern
+
+    def test_dynamic_rank_lora_config(self, lokr_model):
+        # with a dynmaic rank, we expect rank_pattern and alpha_pattern to be set
+        lora_config, state_dict = convert_to_lora(lokr_model, rank=0.5)
+        assert lora_config.r == 1  # dummy value
+        assert lora_config.lora_alpha == 1  # dummy value
+        assert lora_config.rank_pattern
+        assert lora_config.alpha_pattern
+
+        # rank and alpha are always the same, i.e. scaling is 1
+        assert lora_config.rank_pattern == lora_config.alpha_pattern
+        # for each module, two LoRA weights
+        assert 2 * len(lora_config.rank_pattern) == len(state_dict)
+
+    def test_dynamic_rank_1_lora_config(self, lokr_model):
+        # with a dynmaic rank, we expect rank_pattern and alpha_pattern to be set
+        lora_config, state_dict = convert_to_lora(lokr_model, rank=1.0)
+        assert lora_config.r == 1  # dummy value
+        assert lora_config.lora_alpha == 1  # dummy value
+        assert lora_config.rank_pattern
+        assert lora_config.alpha_pattern
+
+        # rank and alpha are always the same, i.e. scaling is 1
+        assert lora_config.rank_pattern == lora_config.alpha_pattern
+        # for each module, two LoRA weights
+        assert 2 * len(lora_config.rank_pattern) == len(state_dict)
+
+    def test_threshold_wrong_value_raises(self, lokr_model):
+        # if a threshold is used, it must be between 0 and 1
+        msg = "If rank is a float, it is interpreted as a threshold. It must be between 0 and 1 but got 123.0"
+        with pytest.raises(ValueError, match=msg):
+            convert_to_lora(lokr_model, rank=123.0)
+
+        msg = "If rank is a float, it is interpreted as a threshold. It must be between 0 and 1 but got -0.5"
+        with pytest.raises(ValueError, match=msg):
+            convert_to_lora(lokr_model, rank=-0.5)
+
+    def test_rank_higher_than_weight_dim_raises(self, lokr_model):
+        # if the requested rank is higher than the weight dimension, we should raise
+        msg = re.escape("The chosen rank 123 is larger than the weight shape (16), please choose a lower rank")
+        with pytest.raises(ValueError, match=msg):
+            convert_to_lora(lokr_model, rank=123)
+
+    def test_fixed_rank_0_raises(self, lokr_model):
+        msg = "Passing a rank of 0 doesn't make sense"
+        with pytest.raises(ValueError, match=msg):
+            convert_to_lora(lokr_model, rank=0)
+
+    def test_converting_transformers_model_works(self, lokr_model, tmp_path):
```

Comment:
Check for a `diffusers` model too?

Reply:
Yes, can do. Do you have a suggestion for a small model that can be tested? As I haven't add conv support yet, it would have to be solely linear layers for now. Or we can add it later, once conv layer support is there?